scalding-serialization/src/main/scala/com/twitter/scalding/serialization/OrderedSerialization.scala

```diff
@@ -101,6 +101,30 @@ object OrderedSerialization {
     case NonFatal(e) => CompareFailure(e)
   }
 
+  def viaTransform[T, U](
```

Comment:
Actually avoiding allocations and nerding in these paths makes sense, i'll just add it and the Injection one in scalding core